phases/19-capstone-projects/47-checkpoint-save-resume/code/main.py

```
@@ -0,0 +1,495 @@
+        m2, o2, s2, loss_fn, loaded,
+        stop_step=total_steps,
+        batches_per_epoch=batches_per_epoch,
+        batch_size=batch_size,
+        in_dim=in_dim,
+        out_dim=out_dim,
+    )
+    resumed_losses = list(loaded.losses)
+
+    suffix_full = full_losses[interrupt_at:]
+    suffix_resumed = resumed_losses[interrupt_at:]
+    max_diff = max(abs(a - b) for a, b in zip(suffix_full, suffix_resumed))
```

Ruff (0.15.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)